### PR TITLE
ASC-1445 output conforms to xsd

### PR DESCRIPTION
**Pull request by zreichert**
- adds config file
- adds global properties to <testsuite>

### CODE REVIEW

**Comment by zreichert** [comment]
This now installs and can run itself!
Here is an example of its use and processing the xml up to the properties.  


(foobaz) MVW10EG8WL:tempest-zigzag zach2872$ export TEMPEST_TEST_CYCLE='I am a custom test cycle'
(foobaz) MVW10EG8WL:tempest-zigzag zach2872$ export TEMPEST_PROJECT_ID=71096
(foobaz) MVW10EG8WL:tempest-zigzag zach2872$ export TEMPEST_BUILD_URL='http://www.lolcats.com'
(foobaz) MVW10EG8WL:tempest-zigzag zach2872$ export TEMPEST_BUILD_NUMBER=23
(foobaz) MVW10EG8WL:tempest-zigzag zach2872$ clear
(foobaz) MVW10EG8WL:tempest-zigzag zach2872$ export TEMPEST_BUILD_URL='http://www.lolcats.com'
(foobaz) MVW10EG8WL:tempest-zigzag zach2872$ export TEMPEST_BUILD_NUMBER=23
(foobaz) MVW10EG8WL:tempest-zigzag zach2872$ export TEMPEST_PROJECT_ID=71096
(foobaz) MVW10EG8WL:tempest-zigzag zach2872$ export TEMPEST_TEST_CYCLE='I am a custom test cycle'
(foobaz) MVW10EG8WL:tempest-zigzag zach2872$ tempest-zigzag --help
Usage: tempest-zigzag [OPTIONS] JUNIT_INPUT_FILE TEST_LIST CONFIG_FILE

  Process multiple files created by tempest into a single accurate junit xml
  artifact

Options:
  --help  Show this message and exit.
(foobaz) MVW10EG8WL:tempest-zigzag zach2872$ tempest-zigzag ~/Desktop/tempest_results_20180912-135147.xml ~/Desktop/tempest_list_tests.txt tempest_zigzag/data/configs/example_tempest-zigzag_config.json
<testsuite failures="2" tests="294" errors="7" name="xml suite created by tempest-zigzag" time="1284.226"><properties><property name="TEMPEST_TEST_CYCLE" value="I am a custom test cycle"/><property name="TEMPEST_BUILD_URL" value="http://www.lolcats.com"/><property name="TEMPEST_PROJECT_ID" value="71096"/><property name="TEMPEST_BUILD_NUMBER" value="23"/></properties><testcase classname="tempest.api.compute.images.test_images.ImagesTestJSON" name="test_delete_saving_image[id-aa06b52b-2db5-4807-b218-9441f75d74e3]" time="18.794"><properties><property name="start_time" value="2019-02-25T19:59:05Z"/><property name="end_time" value="2019-02-25T19:59:23Z"/><property name="test_id" value="aa06b52b-2db5-4807-b218-9441f75d74e3"/><property name="test_step" value="false"/></properties></testcase>

**Comment by zreichert** [comment]
https://rackspace.qtestnet.com/p/71096/portal/project#tab=testexecution&object=4&id=2237457 
That is an example of the newest version of zigzag performing the upload

File: tests/test_tempest_zigzag.py [hunk 1]
```diff
@@ -17,21 +17,27 @@ def test_nothing_to_do(self, file_test_xml_all_pass, file_test_list):
...
         when a setUpClass failure is found"""
 
-        result = TempestZigZag.process_xml(file_test_xml_setup_failure, file_test_list_with_whitespace)
+        result = TempestZigZag.process_xml(file_test_xml_setup_failure,
+                                           file_test_list_with_whitespace,
+                                           tempest_config_file)
         observed = etree.XML(result)
         new_case_count = 0
 
-        assert len(observed) is 10
```
Comment (mwbutcher) [inline tests/test_tempest_zigzag.py:34]: Why did you switch from using 'is' to '==' for the assertions in these tests, just curious?

File: tempest_zigzag/tempest_zigzag.py [hunk 2]
```diff
@@ -1,26 +1,32 @@
...
             test_list_path: (str) the path to the test-list file
+            config_file: (str) the path to the tempest-zigzag configfile
 
         Returns:
             str: the processed xml string
         """
         xml_suite = TempestJunitXMLSuite(junit_input_file_path)
         test_list = TempestTestList(test_list_path)
 
+        #  import pdb; pdb.set_trace()
```
Comment (mwbutcher) [inline tempest_zigzag/tempest_zigzag.py]: breakpoint

File: tests/test_tempest_zigzag.py (hunk 1 above)
Comment (zreichert) [reply tests/test_tempest_zigzag.py:34]: pep8 now distinguishes what you are comparing, It now longer likes it when you use. `is` for comparison of integers.  🤷‍♂️

File: tempest_zigzag/tempest_zigzag.py (hunk 2 above)
Comment (zreichert) [reply tempest_zigzag/tempest_zigzag.py]: oops

File: tests/test_tempest_zigzag.py (hunk 1 above)
Comment (mwbutcher) [reply tests/test_tempest_zigzag.py:34]: :+1:

File: tempest_zigzag/tempest_zigzag.py (hunk 2 above)
Comment (mwbutcher) [reply tempest_zigzag/tempest_zigzag.py]: :+1:

**Review by mwbutcher — APPROVED** [review]

File: constraints.txt [hunk 3]
```diff
@@ -1 +1,2 @@
 pip<10.0.0
+rpc-zigzag~=1.0.0
```
Comment (ryan-rs) [inline constraints.txt]: This pin is too tight.

```suggestion
rpc-zigzag~=1.0
```

File: setup.py [hunk 4]
```diff
@@ -7,7 +7,9 @@
 
 requirements = ['Click>=6.0',
                 'lxml',
-                'future']
+                'future',
+                'jsonschema',
+                'rpc-zigzag~=1.0.0']
```
Comment (ryan-rs) [inline setup.py]: This pin is too tight.

```suggestion
                'rpc-zigzag~=1.0']
```

**Review by ryan-rs — CHANGES_REQUESTED** [review]
Minor changes.

**Review by ryan-rs — APPROVED** [review]

**Pull request merged**